corehq/messaging/scheduling/async_handlers.py

```diff
@@ -1,16 +1,38 @@
 from __future__ import absolute_import
 import json
+from corehq.apps.app_manager.dbaccessors import get_app_ids_in_domain, get_latest_released_app
+from corehq.apps.app_manager.models import Application, AdvancedForm
 from corehq.apps.casegroups.dbaccessors import search_case_groups_in_domain
 from corehq.apps.es import GroupES
 from corehq.apps.hqwebapp.async_handler import BaseAsyncHandler
 from corehq.apps.hqwebapp.encoders import LazyEncoder
 from corehq.apps.locations.models import SQLLocation
 from corehq.apps.users.analytics import get_search_users_in_domain_es_query
+from corehq.util.quickcache import quickcache
 from django.utils.translation import ugettext as _
 
 
+@quickcache(['domain', 'timestamp'], timeout=10 * 60)
+def get_visit_scheduler_forms(domain, timestamp):
+    """
+    The timestamp is set once at the beginning of each loading
+    of the page, so that this result is only calculated once
+    per page load.
+    """
+    result = []
+    for app_id in get_app_ids_in_domain(domain):
+        app = get_latest_released_app(domain, app_id)
+        if isinstance(app, Application):
```

Comment:
Is this intending to filter out linked apps & remote apps?

Reply:
updated https://github.com/dimagi/commcare-hq/pull/19409/commits/bcec3c7107a0913bd01c3a547f50fe2a749780b2